### PR TITLE
[Backend] Rerank submissions In Progress on My Submissions page

### PR DESCRIPTION
**Pull request by Ram81**
### Description

Add custom sorting for submissions with status `RUNNING` `SUBMITTED` or `SUBMITTING` for `My submissions` page. Sorting logic (for submitting/submitted/running status challenge submissions -> sort by created_at asc -> return rank of current submission)

@RishabhJain2018 @deshraj

### CODE REVIEW

File: apps/jobs/utils.py [hunk 1]
```diff
@@ -413,3 +413,58 @@ def get_leaderboard_data_model(submission_pk, challenge_phase_split_pk):
...
+    """
+        Convert a cmp= function into a key= function'
+
+        Arguments:
+             comparator {[function]} -- comparator function
+
+        Returns:
+            [class] -- key class object for lamdbda
+    """
+    class K:
```
Comment (RishabhJain2018) [inline apps/jobs/utils.py:449]: Can we please add a meaningful name here.

File: apps/jobs/views.py [hunk 2]
```diff
@@ -160,8 +161,12 @@ def challenge_submission(request, challenge_id, challenge_phase_id):
         filtered_submissions = SubmissionFilter(
             request.GET, queryset=submission
         )
+        reordered_submissions = sorted(filtered_submissions.qs,
```
Comment (RishabhJain2018) [inline apps/jobs/views.py:164]: Can we please add a comment here with the logic.

File: apps/jobs/utils.py [hunk 3]
```diff
@@ -413,3 +413,58 @@ def get_leaderboard_data_model(submission_pk, challenge_phase_split_pk):
...
+        def __init__(self, obj, *args):
+            self.obj = obj
+
+        def __lt__(self, other):
```
Comment (RishabhJain2018) [inline apps/jobs/utils.py:453]: I didn't get what `__lt__` means. It would be nice if we can have some meaningful names.

Comment (Ram81) [reply apps/jobs/utils.py:453]: `__lt__` is for less than comparison. I don't think we can change comparator key function names, I'll look at it once

File: apps/jobs/utils.py [hunk 4]
```diff
@@ -413,3 +413,64 @@ def get_leaderboard_data_model(submission_pk, challenge_phase_split_pk):
         challenge_phase_split__pk=challenge_phase_split_pk,
     )
     return leaderboard_data
+
+
+def reorder_submissions_comparator(submission1, submission2):
```
Comment (RishabhJain2018) [inline apps/jobs/utils.py:418]: Minor change here -> `submission_1` and `submission_2`.

**Comment by RishabhJain2018** [comment]
@Ram81 Can you please fix the build here?

**Pull request merged**